core/modules/qdisp/Executive.cc

```diff
@@ -470,7 +463,6 @@ void Executive::_waitUntilEmpty() {
         count = _requesters.size();
         _reapRequesters(lock);
         if(count != lastCount) {
-            lastCount = count;
```

Comment:
I think Fabrice is pointing out that you removed the update to lastCount from the loop, so the conditional is always true--it always acts as if the count is new information.